### ISSUE TITLE
Pandas iloc missing overloads

### ISSUE THREAD

**Issue by dcervenkov** [microsoft/python-type-stubs#31]
## Environment data

-   Language Server version: 2020.12.0
-   OS and version: Ubuntu 18.04
-   Python version: 3.6.9

## Expected behaviour

No problem reported

## Actual behaviour

Pylance reports:
```
No overloads for "df.iloc[, 0]" match parameters
  Calling function "__getitem__"
  Argument types: (tuple[slice, Literal[0]])Pylance (reportGeneralTypeIssues)
```

## Code Snippet / Additional information

```python
import pandas as pd
df = pd.DataFrame({"A": [1, 2, 3], "B": [5, 6, 7]})
first_column = df.iloc[:, 0]
```

**Comment by gzheng92** [comment]
I'm having the same issue in the Insider build.  Here's a MWE:

```
import pandas as pd
df = pd.DataFrame({"a_col": list(range(10)), "a_nother": list(range(10)), "b_col": list(range(10))})

df.loc[:, lambda df: df.columns.str.startswith("a_")]
```

The message is:

> No overloads for "__getitem__" match the provided arguments
>   Argument types: (tuple[slice, (df: Unknown) -> Unknown])

Adding the following doesn't improve things:
```
def filter_func(df:pd.DataFrame) -> pd.Series:
    return df.columns.str.startswith("a_")

df.loc[:, filter_func]
```

Here the message is:

> No overloads for "__getitem__" match the provided arguments
>   Argument types: (tuple[slice, (df: DataFrame) -> Series[Unknown]])

**Comment by jakebailey** [comment]
No, that wouldn't help; we need to expand the overloads to also accept callables that return an index as well as the index itself.

**Comment by teneon** [comment]
I ran into the exact same problem today. I wanted to reverse pandas DataFrame like suggested. It was suppose to be one of the fastest ways to do it as well.
```
ds = ds[::-1]
```

I also get this error message: No overloads for "__getitem__" match the provided arguments  Argument types: (slice)

Hopefully this gets fixed :P

**Comment by Dr-Irv** [comment]
tested and fixed in #156 @gramster to close
